ndd_slurm.py

```diff
@@ -56,16 +57,23 @@ def get_master_ndd_cmd(args):
 def get_slave_ndd_cmd(args, env):
     cmd = ['ndd', '-o', args.o]
     slaves = args.S.split(',')
-    index = int(env.get('SLURM_NODEID'))
+    try:
+        current_host = socket.gethostname()
+        index = slaves.index(current_host)
```

Comment:
We should also check every slave hostname for being a prefix of `current_host` as there are now slaves that report full hostname with domain instead of short one. Also please add a warning saying that the current host has been matched by the full hostname, so we can check for it in logs.